app/services/segment_delete_service.rb

```diff
@@ -0,0 +1,88 @@
+# frozen_string_literal: true
+
+class SegmentDeleteService
+  SEGMENT_URI = URI('https://gdpr.segment.com/graphql')
+
+  class SegmentDeleteServiceError < StandardError
+    def initialize(message:, response:)
+      super(message)
+      @response = {body: response.body, code: response.code}
+    end
+
+    attr_reader :response
+
+    def error_type
+      :delete_service
+    end
+  end
+  class SegmentAuthenticationError < SegmentDeleteServiceError
+    def initialize(message: 'Cannot authenticate', response:)
+      super(message: message, response: response)
+    end
+
+    def error_type
+      :authentication
+    end
+  end
+  class SegmentUserDeletionError < SegmentDeleteServiceError
+    def initialize(message: 'Cannot delete user', response:)
+      super(message: message, response: response)
+    end
+
+    def error_type
+      :user_deletion
+    end
+  end
+
+
+  def initialize(user_id)
+    @user_id = user_id
+    @errors = ActiveModel::Errors.new(self)
+  end
+
+  attr_reader :user_id, :errors
+
+  def call!
+    return true unless Features::SegmentDeletionConfig.enabled?
+    request_body = {query: "mutation { createWorkspaceRegulation(workspaceSlug: \"3scale\" type: SUPPRESS_AND_DELETE userId: \"#{user_id}\" ) { id } }"}.to_json
```

Comment:
This `"3scale"` should be configurable

Reply:
You can also use heredoc  for better readability and use string substitution instead of interpolation